src/odemis/driver/xt_client.py

```diff
@@ -980,6 +980,71 @@ def __init__(self, name, role, parent, hfw_nomag, **kwargs):
 
+        pitch_info = self.parent.pitch_info()
+        # TODO change VA and method names with pitch to delta pitch
+        self.pitch = model.FloatContinuous(
+            self.parent.get_pitch()  * 1e-6,
+            unit=pitch_info["unit"],
+            range=tuple(i * 1e-6 for i in pitch_info["range"]),
+            setter=self._setPitch,
+        )
+
+        beam_stigmator_info = self.parent.primary_stigmator_info()
+        beam_stigmator_range_x = beam_stigmator_info["range"]["x"]
+        beam_stigmator_range_y = beam_stigmator_info["range"]["y"]
+        beam_stigmator_range = tuple((i, j) for i, j in zip(beam_stigmator_range_x, beam_stigmator_range_y))
+        self.beamStigmator = model.TupleContinuous(
+            tuple(self.parent.get_primary_stigmator()),
+            unit=beam_stigmator_info["unit"],
+            range=beam_stigmator_range,
+            setter=self._setBeamStigmator)
+
+        pattern_stigmator_info = self.parent.secondary_stigmator_info()
+        pattern_stigmator_range_x = pattern_stigmator_info["range"]["x"]
+        pattern_stigmator_range_y = pattern_stigmator_info["range"]["y"]
+        pattern_stigmator_range = tuple((i, j) for i, j in zip(pattern_stigmator_range_x,
+                                                                   pattern_stigmator_range_y))
+        self.patternStigmator = model.TupleContinuous(
+            tuple(self.parent.get_secondary_stigmator()),
+            unit=pattern_stigmator_info["unit"],
+            range=pattern_stigmator_range,
+            setter=self._setPatternStigmator)
+
+        self.beamShiftTransformtionMatrix = model.ListVA(
+            self.parent.get_dc_coils(),
+            unit=None,
+            readonly=True)
+
+        self.multiprobeRotation = model.FloatVA(
+            numpy.deg2rad(self.parent.get_mpp_orientation()),
+            unit="radians",
+            readonly=True)
+
+        aperture_index_info = self.parent.aperture_index_info()
+        self.apertureIndex = model.IntContinuous(
+            int(self.parent.get_aperture_index()),
+            unit=aperture_index_info["unit"],
+            range=tuple(int(i) for i in aperture_index_info["range"]),
+            setter=self._setApertureIndex)
+
+        beamlet_index_info = self.parent.beamlet_index_info()
+        beamlet_index_range_x = beamlet_index_info["range"]["x"]
+        beamlet_index_range_y = beamlet_index_info["range"]["y"]
+        beamlet_index_range = tuple((int(i), int(j)) for i, j in zip(beamlet_index_range_x, beamlet_index_range_y))
+        beamlet_index = self.parent.get_beamlet_index()
+        self.beamletIndex = model.TupleContinuous(
+            tuple(int(i) for i in beamlet_index),  # convert tuple values to integers.,
+            unit=beamlet_index_info["unit"],
+            range=beamlet_index_range,
+            setter=self._setBeamletIndex
+        )
+
+        beam_mode = True if self.parent.get_use_case() == 'MultiBeamTile' else False
```

Comment:
You can directly write `beam_mode = (self.parent.get_use_case() == 'MultiBeamTile')`.
(please update also update_settings() function)

BTW, could you rename `beam_mode` to `mbeam_mode` or `multibeam_mode`? Just "beam_mode" sounds like it should be some kind of type of beam, not really just a boolean.